packages/eslint-plugin-prairielearn/src/rules/aws-client-mandatory-config.ts

```diff
@@ -28,19 +28,21 @@ export default ESLintUtils.RuleCreator.withoutDocs({
         clientNames.forEach((clientName) => awsClientImports.add(clientName));
       },
       NewExpression(node) {
-        if (node.callee.type === 'Identifier' && awsClientImports.has(node.callee.name)) {
+        if (
+          node.callee.type === 'Identifier' &&
+          awsClientImports.has(node.callee.name) &&
+          node.arguments.length === 0
           // We're constructing an AWS client. Ensure that the call has at
           // least one argument corresponding to a config object.
```

Comment:
Were these auto-fixed? I don't think it's doing a good job of putting comments on the correct line. Apologies if you were aware of this and going to fix this manually before merging.

Related: I actually think nested `if` blocks like this are a useful as a narrative feature precisely because they give one convenient places to place comments like this. One could put the comment e.g. above the `node.arguments.length ...` line, but then you have to mentally evaluate the short circuiting.

Reply:
Reverted in 2608926f1b0a7cf00420466b7744fbd17e6b11e1.